### ISSUE TITLE
empty redis_url string configuration for scheduled server-side syncs 

### ISSUE THREAD

**Issue by chaserx** [octobox/octobox#639]
Are you experiencing this issue in octobox.io or your own instance?
 - [ ] octobox.io
 - [x] My instance

🐞 **Problem**
I was following the instructions for `Scheduling server-side notification syncs` via Option 2 by including `OCTOBOX_SIDEKIQ_SCHEDULE_ENABLED` in my .env file. Strangely, the app would not boot because of a `invalid uri scheme ''` error that could be traced back to [Line 14 of the sidekiq.rb](https://github.com/octobox/octobox/blob/master/config/initializers/sidekiq.rb#L14) initializer. 

It took me a bit to get it, but I think it's because the [redis_url is nil](https://github.com/mperham/sidekiq/issues/3095#issuecomment-238283509). 

🎯 **Goal**
Using sidekiq-scheduler for scheduled background syncs. 

💡 **Possible solutions**

I have a possible solution that I can post in a PR soon, but I'll describe what I've done below. 

Basically, following the configuration option shown in the sidekiq-scheduler docs seems to work and the error is resolved. 

📋  **Steps to solve the problem**

```shell
redis.service.octobox.internal_1     | 1:M 28 May 23:08:47.771 * The server is now ready to accept connections on port 6379
app_1                                | rake aborted!
app_1                                | ArgumentError: invalid uri scheme ''
app_1                                | /usr/local/bundle/gems/redis-4.0.1/lib/redis/client.rb:415:in `_parse_options'
app_1                                | /usr/local/bundle/gems/redis-4.0.1/lib/redis/client.rb:79:in `initialize'
app_1                                | /usr/local/bundle/gems/redis-4.0.1/lib/redis.rb:38:in `new'
app_1                                | /usr/local/bundle/gems/redis-4.0.1/lib/redis.rb:38:in `initialize'
app_1                                | /usr/local/bundle/gems/sidekiq-5.1.3/lib/sidekiq/redis_connection.rb:55:in `new'
app_1                                | /usr/local/bundle/gems/sidekiq-5.1.3/lib/sidekiq/redis_connection.rb:55:in `build_client'
app_1                                | /usr/local/bundle/gems/sidekiq-5.1.3/lib/sidekiq/redis_connection.rb:34:in `block in create'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool/timed_stack.rb:171:in `try_create'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool/timed_stack.rb:83:in `block (2 levels) in pop'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool/timed_stack.rb:79:in `loop'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool/timed_stack.rb:79:in `block in pop'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool/timed_stack.rb:78:in `synchronize'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool/timed_stack.rb:78:in `pop'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool.rb:93:in `checkout'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool.rb:62:in `block in with'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool.rb:61:in `handle_interrupt'
app_1                                | /usr/local/bundle/gems/connection_pool-2.2.2/lib/connection_pool.rb:61:in `with'
app_1                                | /usr/local/bundle/gems/sidekiq-5.1.3/lib/sidekiq.rb:92:in `redis'
app_1                                | /usr/local/bundle/gems/sidekiq-scheduler-2.2.2/lib/sidekiq-scheduler/redis_manager.rb:99:in `schedule_exist?'
app_1                                | /usr/local/bundle/gems/sidekiq-scheduler-2.2.2/lib/sidekiq-scheduler/schedule.rb:84:in `get_all_schedules'
app_1                                | /usr/local/bundle/gems/sidekiq-scheduler-2.2.2/lib/sidekiq-scheduler/schedule.rb:43:in `schedule='
app_1                                | /usr/src/app/config/initializers/sidekiq.rb:14:in `<main>'
app_1                                | /usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:50:in `load'
app_1                                | /usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:50:in `load'
app_1                                | /usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:277:in `block in load'
app_1                                | /usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
app_1                                | /usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:277:in `load'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/engine.rb:657:in `block in load_config_initializer'
app_1                                | /usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/notifications.rb:170:in `instrument'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/engine.rb:656:in `load_config_initializer'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/engine.rb:614:in `block (2 levels) in <class:Engine>'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/engine.rb:613:in `each'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/engine.rb:613:in `block in <class:Engine>'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/initializable.rb:32:in `instance_exec'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/initializable.rb:32:in `run'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/initializable.rb:61:in `block in run_initializers'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/initializable.rb:50:in `each'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/initializable.rb:50:in `tsort_each_child'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/initializable.rb:60:in `run_initializers'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/application.rb:361:in `initialize!'
app_1                                | /usr/src/app/config/environment.rb:5:in `<main>'
app_1                                | /usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
app_1                                | /usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
app_1                                | /usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
app_1                                | /usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
app_1                                | /usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
app_1                                | /usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
app_1                                | /usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
app_1                                | /usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/application.rb:337:in `require_environment!'
app_1                                | /usr/local/bundle/gems/railties-5.2.0/lib/rails/application.rb:520:in `block in run_tasks_blocks'
app_1                                | /usr/local/bundle/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
app_1                                | /usr/local/bundle/bin/bundle:104:in `load'
app_1                                | /usr/local/bundle/bin/bundle:104:in `<main>'
app_1                                | Tasks: TOP => db:migrate => db:load_config => environment
app_1                                | (See full trace by running task with --trace)
octobox_app_1 exited with code 1
```

What's working for me is to move the scheduler configuration code up from the bottom of that file into `configure_server` block. Like so, 

```ruby
Sidekiq.configure_server do |config|
  config.redis = { url: Octobox.config.redis_url }
  if Octobox.config.sidekiq_schedule_enabled?
    config.on(:startup) do
      Sidekiq.schedule = YAML.load_file(Octobox.config.sidekiq_schedule_path)
      Sidekiq::Scheduler.reload_schedule!
    end
  end
end
```

🎉

**Comment by chaserx** [comment]
this does solve the error, but sidekiq is not running with the current docker-compose setup for development as far as I can tell.

**Comment by andrew** [comment]
@chaserx thanks for investigating this, yeah we should update the docker-compose setup to get sidekiq going, I think we'll end up enabling the sidekiq syncing by default sooner rather than later.

**Comment by chaserx** [comment]
closed by #640
